src/Amplitude.js

```diff
@@ -145,6 +151,15 @@ var constructor = function () {
 
     function onUserIdentified(user) {
         var userIdMissing;
+
+        function setUserId(userId) {
+            if (userId) {
+                return getInstance().setUserId(userId);
+            } else {
+                userIdMissing = true;
```

Comment:
Out of curiosity, instead checking for a `userIdMissing` boolean, why not just check to see if `userId` is `undefined` or `null`?

Reply:
good call.  this can be simplified considerably